spec/lib/rdstation/client_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+RSpec.describe RDStation::Client do
+  context "when access_token is given" do
+    let(:access_token) { 'access_token' }
+    let(:client) { described_class.new(access_token: access_token) }
+
+    it "returns the correct endpoint" do
+      expect(client.contacts).to be_a(RDStation::Contacts)
+      expect(client.events).to be_a(RDStation::Events)
+      expect(client.fields).to be_a(RDStation::Fields)
+      expect(client.webhooks).to be_a(RDStation::Webhooks)
+    end
+    
+    it "creates an authorization_header and initilizes the enpoints with this header" do
+      mock_authorization_header = double(RDStation::AuthorizationHeader)
+      expect(RDStation::AuthorizationHeader).to receive(:new)
+        .with({ access_token: access_token})
+        .and_return(mock_authorization_header)
+      expect(RDStation::Contacts).to receive(:new).with({ authorization_header: mock_authorization_header })
+      expect(RDStation::Events).to receive(:new).with({ authorization_header: mock_authorization_header })
+      expect(RDStation::Fields).to receive(:new).with({ authorization_header: mock_authorization_header })
+      expect(RDStation::Webhooks).to receive(:new).with({ authorization_header: mock_authorization_header })
+      client.contacts
+      client.events
+      client.fields
+      client.webhooks
```

Comment:
Também não impeditivo: como sugestão de organização acho que esse spec ficaria melhor quebrando em um `it` pra cada endpoint. O mock poderia ficar em um `let` dentro do contexto. 

Ainda uma outra idéia, poderia ser esperado retornar o objeto do endpoint correto pra cada método. Algo assim:

```
let(:mock_authorization_header) { double(RDStation::AuthorizationHeader) }
before {  allow(RDStation::AuthorizationHeader).to receive(:new).and_return mock_authorization_header }

it 'returns Contacts endpoint handler or sumthing' do
  expect(RDStation::Contacts).to receive(:new).with({ authorization_header: mock_authorization_header })
  expect(client.contacts).to be_instance_of RDStation::Contacts
end
```

Reply:
Cadastrei como débito técnico https://github.com/ResultadosDigitais/rdstation-ruby-client/issues/34